### DISCUSSION TITLE
What heppened to Fermi?

### DISCUSSION THREAD

**Discussion by serzhiio** [DioxusLabs/dioxus#1927] [Q&A] [answered]
Any updates i missed?
```
[serzh@serzh web_ui_new]$ cargo update
    Updating crates.io index
    Updating git repository `https://github.com/DioxusLabs/dioxus`
error: no matching package named `fermi` found
```

**Comment by ealmloff** [comment]
Fermi atoms are now `Signal::global`. I finished the migration guide today: https://github.com/ealmloff/docsite/blob/dioxus-next/docs-src/0.4/en/migration/fermi.md. You will need to serve the docsite yourself with `dx serve --platform fullstack`

[fermi migration.pdf](https://github.com/DioxusLabs/dioxus/files/14273879/fermi.migration.pdf)

**Comment by serzhiio** [comment]
Here is how docs looks:
![Снимок экрана от 2024-02-14 13-36-18](https://github.com/DioxusLabs/dioxus/assets/10449591/b1c2552e-4b7d-4623-b361-1b892625af0b)

**Reply by ealmloff** [reply]
That is what the markdown looks like before it is rendered by dioxus. The pdf I uploaded has the rendered version. Some rust snippets are external so we can check them with `cargo check` and format them with `cargo fmt`

**Comment by serzhiio** [comment]
And how is subscription context managed with GlobalSignals?
If in one context i only write will it be subscribed to updates?
By context i mean component.

**Comment by serzhiio** [comment]
>Signals will only subscribe to components when you read from the signal in that component. It will never subscribe to a component when reading data in a future or event handler.

If this is how things really done, no more questions. Sounds great!

**Reply by ealmloff** [reply, marked as answer]
Yes, that applies to global signals as well.

**Reply by serzhiio** [reply]
Very nice API, i am just discovered `.peek()` method! But still can't understand the `_memo` usecases.
